runtime/doc/lua-guide.txt

```diff
@@ -0,0 +1,311 @@
+*lua-guide.txt*	Nvim
+
+			    NVIM REFERENCE MANUAL
+
+			  Guide to using Lua in Nvim
+
+
+This chapter will give you the basics of using Lua in Neovim.
+
+				       Type |gO| to see the table of contents.
+
+==============================================================================
+Introduction							     *lua-guide*
+
+COMMON QUESTIONS
+
+Maybe this section isn't needed. We'll see.
+
+- Is vimscript being deprecated?
+
+Nope. Vimscript is being promoted. Vim9script however is deprecated.
+
+- Is my configuration gonna be faster?
+
+Probably not.
+
+- Explain difference between api vs lua api.
+ api for long-term guarantees it'll work, lua api for convenience. Plugins
+ should probably use api and configs should probably use lua api
+
+Probably way more, blanking out at the moment.
+
+==============================================================================
+Using Lua from Neovim
+
+You can use `:lua` to run Lua code. >
+
+	:lua print("Hello!")
+
+You can also use `:lua=`, which is same as `:lua print(vim.inspect(expr))`, to
+conveniently check the value of a variable or a table. >
+
+	:lua= package
+
+- Mention heredocs?
+- putting Lua files in rtp
+- :source vs. :luafile vs. require
+
+==============================================================================
+Calling Vimscript from Lua
+- vim.cmd
+- vim.fn
+- vim.api
+
+==============================================================================
+Setting variables
+- vim.g
+- vim.b
+- vim.w
+- vim.t
+- vim.v
+- vim.env
+
+==============================================================================
+Thermonuclear disaster/Setting options
+- vim.opt
+- vim.opt_global
+- vim.opt_local
+	- get()
+	- append()
+	- prepend()
+	- remove()
+- vim.o
+- vim.bo
+- vim.wo
+- vim.go
+
+==============================================================================
+Defining mappings
+
+TODO: not a fan of both the API and lua, yeet the API
+
+API FUNCTIONS
+
+Neovim provides a list of API functions to set, get and delete mappings:
+
+- Global mappings:
+    - `vim.api.nvim_set_keymap()`
+    - `vim.api.nvim_get_keymap()`
+    - `vim.api.nvim_del_keymap()`
+- Buffer-local mappings:
+    - `vim.api.nvim_buf_set_keymap()`
+    - `vim.api.nvim_buf_get_keymap()`
+    - `vim.api.nvim_buf_del_keymap()`
+
+Let's start with `vim.api.nvim_set_keymap()` and
+`vim.api.nvim_buf_set_keymap()`
+
+The first argument passed to the function is a string containing the name of
+the mode for which the mapping will take effect:
+
+| String value           | Help page     | Affected modes                           | Vimscript equivalent |
+| ---------------------- | ------------- | ---------------------------------------- | -------------------- |
+| `''` (an empty string) | `mapmode-nvo` | Normal, Visual, Select, Operator-pending | `:map`               |
```

Comment:
Double quotes instead of single quotes, `||` for tags
```suggestion
| "" (an empty string)   | |mapmode-nvo| | Normal, Visual, Select, Operator-pending | |:map|               |
```